File: java/classes/inheritance/ClassAccessSpecifiers.java
class Mammal {
  public boolean isDomesticated;

  protected boolean hasTail;

  private boolean isMilkFeeded = true;
  private String classTaxon = "Mammalia";
}

class Fox extends Mammal {
  public String name;

  public void show() {
    System.out.println("Hi, my name is " + name
      // + ".\nClass: " + classTaxon
      + "\nSpecies: " + speciesTaxon
      // + "\nAm I milk-feeded as a cub? " + isMilkFeeded
      + "\nDo I have tail? " + hasTail
      + "\nDo I have fur? " + hasFur
      + "\nAm I domesticated? " + isDomesticated
    );
  };

  private boolean hasFur = true;
  private String speciesTaxon = "Vulpes vulpes";
}

class UrbanFox extends Fox {
  public void display() {
    System.out.println("Hi, my name is " + name
      // + ".\nClass: " + classTaxon
      // + "\nSpecies: " + speciesTaxon
      // + "\nAm I milk-feeded as a cub? " + isMilkFeeded
      + "\nDo I have tail? " + hasTail
      // + "\nDo I have fur? " + hasTur
      + "\nAm I domesticated? " + isDomesticated
    );
  }
}

class ClassAccessSpecifiers {
  public static void main(String[] args) {
    Fox foxFerdinand = new Fox();

    foxFerdinand.name = "Ferdinand";
    foxFerdinand.isDomesticated = true;

    foxFerdinand.show();

    System.out.println("\nHi, my name is " + foxFerdinand.name
      // + ".\nClass: " + foxFerdinand.classTaxon
      // + "\nSpecies: " + foxFerdinand.speciesTaxon
      // + "\nAm I milk-feeded as a cub? " + isMilkFeeded
      + "\nDo I have tail? " + foxFerdinand.hasTail
      // + "\nDo I have fur? " + foxFerdinand.hasFur
      + "\nAm I domesticated? " + foxFerdinand.isDomesticated
      + "\n"
    );

    UrbanFox foxMelody = new UrbanFox();

    foxMelody.name = "Melody";

    foxMelody.display();
  }
}

File: c#/classes/inheritance/ClassAccessSpecifiers.cs
class Mammal {
  public bool IsDomesticated;

  protected bool HasTail;

  private bool IsMilkFeeded = true;
  private const string ClassTaxon = "Mammalia";
}

class Fox : Mammal {
  public string Name;

  public void Show() {
    System.Console.WriteLine("Hi, my name is " + Name
      // + ".\nClass: " + ClassTaxon
      + "\nSpecies: " + SpeciesTaxon
      // + "\nAm I milk-feeded as a cub? " + IsMilkFeeded
      + "\nDo I have tail? " + HasTail
      + "\nDo I have fur? " + HasFur
      + "\nAm I domesticated? " + IsDomesticated
    );
  }

  private bool HasFur = true;
  private const string SpeciesTaxon = "Vulpes vulpes";
}

class UrbanFox : Fox {
  public void Display() {
    System.Console.WriteLine("Hi, my name is " + Name
      // + ".\nClass: " + ClassTaxon
      // + "\nSpecies: " + SpeciesTaxon
      // + "\nAm I milk-feeded as a cub? " + IsMilkFeeded
      + "\nDo I have tail? " + HasTail
      // + "\nDo I have fur? " + HasFur
      + "\nAm I domesticated? " + IsDomesticated
    );
  }
}

class ClassAccessSpecifiers {
  public static void Main(string[] args) {
    Fox foxFerdinand = new Fox();

    foxFerdinand.Name = "Ferdinand";
    foxFerdinand.IsDomesticated = true;

    foxFerdinand.Show();

    System.Console.WriteLine("\nHi, my name is " + foxFerdinand.Name
      // + ".\nClass: " + foxFerdinand.ClassTaxon
      // + "\nSpecies: " + foxFerdinand.SpeciesTaxon
      // + "\nAm I milk-feeded as a cub? " + foxFerdinand.IsMilkFeeded
      // + "\nDo I have tail? " + foxFerdinand.HasTail
      // + "\nDo I have fur? " + foxFerdinand.HasFur
      + "\nAm I domesticated? " + foxFerdinand.IsDomesticated
      + "\n"
    );

    UrbanFox foxMelody = new UrbanFox();

    foxMelody.Name = "Melody";

    foxMelody.Display();
  }
}
